### DISCUSSION TITLE
Trying to add Pyodide to CI

### DISCUSSION THREAD

**Discussion by hoodmane** [wjakob/nanobind#694] [General]
I applied this patch to pyproject.toml:
```patch
 [tool.scikit-build]
 minimum-version = "build-system.requires"
-wheel.platlib = false
-cmake.define.NB_TEST = false
+wheel.platlib = true
+cmake.define.NB_TEST = true
+cmake.define.NB_TEST_STABLE_ABI = true
+cmake.define.NB_TEST_SHARED_BUILD = true
```
and now I'm trying to run `cibuildwheel --only=cp312-pyodide_wasm32`. I needed to apply #693. After that it fails here:
```py
[22/94] Generating py_stub_test.pyi
FAILED: tests/py_stub_test.pyi /tmp/tmpqa002vg1/build/tests/py_stub_test.pyi 
cd /tmp/tmpqa002vg1/build/tests && /tmp/build-env-lpoyzm2l/bin/python /home/rchatham/Documents/programming/nanobind/src/stubgen.py -q -i /tmp/tmpqa002vg1/build/tests -m py_stub_test -o py_stub_test.pyi
Traceback (most recent call last):
  <...>
  File "/home/rchatham/Documents/programming/nanobind/src/stubgen.py", line 1346, in main
    mod_imported = importlib.import_module(mod)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  <...>
ModuleNotFoundError: No module named 'py_stub_test'
```
It seems that the `build/tests` directory only contains the following four files: `cmake_install.cmake`, `test_typing.py`, `CMakeFiles`, `libinter_module.s`'.
I hit the same error when I do `cibuildwheel --only cp312-manylinux_x86_64`.

@henryiii any idea what I should do here? Also, is it possible to make a `pyproject.toml` file in the test directory that instructs scikit-build to build from the parent directory? Or to have a second `[tool.scikit-build.test]` set of options for scikit-build?

**Comment by henryiii** [comment]
Yes. For the main directory approach, instead of patching it, you can use `[[tool.scikit-build.overrides]]`.

```toml
[tool.scikit-build]
minimum-version = "build-system.requires"
wheel.platlib = false
cmake.define.NB_TEST = false

[[tool.scikit-build.overrides]]
if.env.BUILD_TESTING = true
wheel.platlib = true
cmake.define.NB_TEST = true
cmake.define.NB_TEST_STABLE_ABI = true
cmake.define.NB_TEST_SHARED_BUILD = true
```

Then you can set `BUILD_TESTING=1` when building to make the tests.

#697 has an example of the test directory approach.
